rand_distr/src/utils.rs

```diff
@@ -82,6 +85,15 @@ impl Float for f32 {
         }
     }
 
+    #[inline]
+    fn to_usize(self) -> Option<usize> {
+        if self >= 0. && self <= ::core::usize::MAX as f32 {
```

Comment:
This is incorrect for 32-bit platforms. Check [this out](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cad442b6fc5cfc87f448cbabc2da9be1).

This isn't your fault; `to_u64` for `f64` makes the same mistake. Hmm, even the `TryFrom` in Rust 1.34 doesn't handle this conversion. I'm starting to hate [this topic](https://github.com/rust-lang/rfcs/pull/2484); seems simple but isn't.

Reply:
Ohh this issue. I don't know how to solve this aswell. The link didn't suggest a solution. I think this belongs in another PR, or something?

Reply:
Yes: #973